web/src/components/storage/ProposalDeviceSection.jsx

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * @typedef {import ("~/client/storage").ProposalManager.ProposalSettings} ProposalSettings
+ * @typedef {import ("~/client/storage").DevicesManager.StorageDevice} StorageDevice
+ * @typedef {import ("~/client/storage").ProposalManager.Volume} Volume
+ */
+
+/**
+ * Form for selecting the installation device.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} props.id - Form ID.
+ * @param {StorageDevice|undefined} [props.current] - Currently selected device, if any.
+ * @param {StorageDevice[]} [props.devices=[]] - Available devices for the selection.
+ * @param {onSubmitFn} [props.onSubmit=noop] - On submit callback.
+ *
+ * @callback onSubmitFn
+ * @param {string} device - Name of the selected device.
+ */
+const InstallationDeviceForm = ({
+  id,
+  current,
+  devices = [],
+  onSubmit = noop
+}) => {
+  const [device, setDevice] = useState(current || devices[0]);
+
+  const changeSelected = (deviceId) => {
+    setDevice(devices.find(d => d.sid === deviceId));
+  };
+
+  const submitForm = (e) => {
+    e.preventDefault();
+    if (device !== undefined) onSubmit(device);
+  };
+
+  return (
+    <Form id={id} onSubmit={submitForm}>
+      <DeviceSelector
+        selected={device}
+        devices={devices}
+        onChange={changeSelected}
+      />
+    </Form>
+  );
+};
+
+/**
+ * Allows to select the installation device.
+ * @component
+ *
+ * @param {object} props
+ * @param {string} [props.current] - Device name, if any.
+ * @param {StorageDevice[]} [props.devices=[]] - Available devices for the selection.
+ * @param {boolean} [props.isLoading=false] - Whether to show the selector as loading.
+ * @param {onChangeFn} [props.onChange=noop] - On change callback.
+ *
+ * @callback onChangeFn
+ * @param {string} device - Name of the selected device.
+ */
+const InstallationDeviceField = ({
+  current,
+  devices = [],
+  isLoading = false,
+  onChange = noop
+}) => {
+  const [device, setDevice] = useState(devices.find(d => d.name === current));
+  const [isFormOpen, setIsFormOpen] = useState(false);
+
+  const openForm = () => setIsFormOpen(true);
+
+  const closeForm = () => setIsFormOpen(false);
+
+  const acceptForm = (selectedDevice) => {
+    closeForm();
+    setDevice(selectedDevice);
+    onChange(selectedDevice);
+  };
+
+  /**
+   * Renders a button that allows changing selected device
+   *
+   * NOTE: if a device is already selected, its name and size will be used for
+   * the button text. Otherwise, a "No device selected" text will be shown.
+   *
+   * @param {object} props
+   * @param {StorageDevice|undefined} [props.current] - Currently selected device, if any.
+   */
+  const DeviceContent = ({ device }) => {
+    return (
+      <Button variant="link" isInline onClick={openForm}>
+        {device ? deviceLabel(device) : _("No device selected yet")}
+      </Button>
+    );
+  };
+
+  if (isLoading) {
+    return <Skeleton screenreaderText={_("Loading selected device")} width="25%" />;
```

Comment:
Note: I realized that we are rendering these loading skeletons without an accessible text. Not sure if the one chosen here is the way to go, but definitely we should think on a pattern for describing all of them.

Reply:
What does not fit to me here is the "Loading" word. It's actually not "loading" such a devices, but _gathering the_ or _waiting for_ information to know which device is selected.

Reply:
I think that "Waiting for information about" is a good pattern for those texts.

Reply:
Thanks a lot!